scripts/build_osm_network.py

```diff
@@ -684,6 +691,9 @@ def fix_overpassing_lines(lines, buses, distance_crs, tol=1):
     # return to original crs
     df_l = df_l.to_crs(lines.crs)
 
+    # remove lines that are rings (included for completion), TODO: this should be a separate function
+    df_l = df_l[~df_l.geometry.is_ring].reset_index()
```

Comment:
there was one line that was a ring after splitting and causing problems with the .boundries function in set_line_id

Reply:
Interesting, a question: was the ring already there or how was it created?

Reply:
I believe the ring structure was there, but it was connected to a line so the shapely boundary function still worked, but once split it becomes exclusively a ring and causes problems.

Reply:
Could be nice to plot that and understand what happens as it may happen more. Potentially we could fix that by replacing the ring with a line maybe

Reply:
Rings are only a problem because of the subsequent function (set_line_id) that gets called that uses the .boundary() function. The .boundary() causes problems with rings and is really an edge case, that should be tested for in the set_line_id. I have provided a one line fix to make the CI pass for now and added comments in the code. fyi, only one ring was removed in the test case, and i don't expect it to be common occurrence.